### DISCUSSION TITLE
Custom policy: yaml selector not working

### DISCUSSION THREAD

**Discussion by merou** [aquasecurity/trivy#7153] [Bugs]
### Description

I am trying to use a simple example from:
github.com/aquasecurity/katacoda-scenarios/Trivy/assets/examples/misconf/custom-policy/serverless

@ana

### Desired Behavior

It is supposed to return an error on the python version used.

### Actual Behavior

No files are detected.

### Reproduction Steps

```bash
1. clone katacoda-scenarios repository
2. cd Trivy/assets/examples/misconf/custom-policy/serverless
3. run trivy conf --config-check ./policy --namespaces all ./configs --debug
```

### Target

Filesystem

### Scanner

Misconfiguration

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
2024-07-12T12:50:17+02:00	DEBUG	Cache dir	dir="/home/user/.cache/trivy"
2024-07-12T12:50:17+02:00	DEBUG	Parsed severities	severities=[UNKNOWN LOW MEDIUM HIGH CRITICAL]
2024-07-12T12:50:17+02:00	INFO	Misconfiguration scanning is enabled
2024-07-12T12:50:17+02:00	DEBUG	Policies successfully loaded from disk
2024-07-12T12:50:17+02:00	DEBUG	Enabling misconfiguration scanners	scanners=[azure-arm cloudformation dockerfile helm kubernetes terraform terraformplan-json terraformplan-snapshot]
2024-07-12T12:50:17+02:00	DEBUG	Initializing scan cache...	type="memory"
2024-07-12T12:50:17+02:00	DEBUG	OS is not detected.
2024-07-12T12:50:17+02:00	INFO	Detected config files	num=0
```

### Operating System

linux

### Version

```bash
$ trivy --version                                                                  
Version: 0.53.0
Check Bundle:
  Digest: sha256:1df8ade71efc830877ca3b1130f83e0c6368e3a45b0d4c0f0418955501644054
  DownloadedAt: 2024-07-12 10:45:18.66132513 +0000 UTC
```

### Checklist

- [X] Run `trivy clean --all`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by nikpivkin** [comment]
Hi @merou !
Currently Trivy does not support scanning `yaml` files. See the [documentation](https://aquasecurity.github.io/trivy/v0.53/docs/coverage/iac/) for a list of supported configurations.

**Reply by merou** [reply]
Hello @nikpivkin thank you for your answer - so it means custom rego policies only applies to the list you provided.
Ok looking forward to an implementation in Trivy - for now I will use conftest.

**Reply by simar7** [reply]
@merou to add to what @nikpivkin mentioned, currently by design Trivy has detection logic for a variety of configurations, e.g. Terraform, Cloudformation, etc. Custom checks today assume that the scanned input is one of supported configuration types.


When it comes to implementing support for unknown file parsing, there are a few things involved:

1. First we have to determine the kind of configuration, as mentioned earlier that list can be found in the [docs](https://aquasecurity.github.io/trivy/v0.53/docs/coverage/iac/).
2. If we determine it isn't one of the supported file configuration types, we have to fallback to a custom schema, if provided by the user.
3. If the user does not provide a custom schema and we are not able to determine the kind of configuration, Trivy is not able to determine the input as something it knows about. 

I will create an issue with these details to track as a feature request.

**Reply by simar7** [reply]
Track https://github.com/aquasecurity/trivy/issues/7158
